src/main/java/uk/gov/hmcts/reform/prl/handlers/TransferToAnotherCourtEventHandler.java

```diff
@@ -0,0 +1,158 @@
+
+        sendEmailToOtherParties(caseData, EmailTemplateNames.TRANSFER_COURT_EMAIL_NOTIFICATION_OTHER_PARTIES);
+    }
+
+    private void sendEmailToRespondents(CaseData caseData, EmailTemplateNames emailTemplateNames) {
+        Map<String, String> respondentEmail = caseData.getRespondents().stream()
+            .map(Element::getValue)
+            .filter(respondent -> !CaseUtils.hasLegalRepresentation(respondent)
+                && Yes.equals(respondent.getCanYouProvideEmailAddress()))
+            .collect(Collectors.toMap(
+                PartyDetails::getEmail,
+                party -> party.getFirstName() + " " + party.getLastName(),
+                (x, y) -> x
+            ));
+
+        if (!respondentEmail.isEmpty()) {
+            respondentEmail.forEach(
+                (key, value) ->
+                    emailService.send(
+                        key,
+                        emailTemplateNames,
+                        transferCaseContentProvider.buildCourtTransferEmailCitizen(caseData, value,false),
+                        LanguagePreference.getPreferenceLanguage(caseData)
+                    ));
+        }
+    }
+
+    private void sendEmailToRespondentSolicitor(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        List<Map<String, List<String>>> respondentSolicitors = caseData
+            .getRespondents()
+            .stream()
+            .map(Element::getValue)
+            .filter(i -> YesNoDontKnow.yes.equals(i.getDoTheyHaveLegalRepresentation()))
+            .map(i -> {
+                Map<String, List<String>> temp = new HashMap<>();
+                temp.put(i.getSolicitorEmail(), List.of(
+                    i.getRepresentativeFirstName() + " " + i.getRepresentativeLastName(),
+                    i.getFirstName() + " " + i.getLastName()
+                ));
+                return temp;
+            })
+            .collect(Collectors.toList());
+
+        for (Map<String, List<String>> resSols : respondentSolicitors) {
+            String solicitorEmail = resSols.keySet().toArray()[0].toString();
+            emailService.send(
+                solicitorEmail,
+                emailTemplateName,
+                transferCaseContentProvider.buildCourtTransferEmailSolicitor(caseData, resSols.get(solicitorEmail).get(0)),
+                LanguagePreference.english
+            );
+        }
+    }
+
+    private void sendEmailToApplicant(CaseData caseData, EmailTemplateNames emailTemplateName) {
+        Map<String, String> applicantEmails = caseData.getApplicants().stream()
+            .map(Element::getValue)
+            .filter(applicant -> !CaseUtils.hasLegalRepresentation(applicant)
+                && Yes.equals(applicant.getCanYouProvideEmailAddress()))
+            .collect(Collectors.toMap(
+                PartyDetails::getEmail,
+                party -> party.getFirstName() + " " + party.getLastName(),
```

Comment:
use the common method

Reply:
used common method instead of individual method to get values